workspaces/analytics/plugins/analytics-module-matomo/src/api/Matomo.ts

```diff
@@ -86,13 +94,20 @@ export class MatomoAnalytics implements AnalyticsApi {
     ]);
   }
 
-  private async setUserFrom(identityApi: IdentityApi) {
+  private async setUserFrom(
+    identityApi: IdentityApi,
+    disableEncryption?: boolean,
+  ) {
     const { userEntityRef } = await identityApi.getBackstageIdentity();
```

Comment:
nothing to do with your changes but does this logic work? When the user isn't logged in, this won't return any `userEntityRef` for the entire session until the page is refreshed 🤔